src/dogroups.c

```diff
@@ -167,6 +174,10 @@ SEXP dogroups(SEXP dt, SEXP dtcols, SEXP groups, SEXP grpcols, SEXP jiscols, SEX
         case REALSXP :
           REAL(VECTOR_ELT(xSD,j))[0] = NA_REAL;
           break;
+        case CPLXSXP :
+          COMPLEX(VECTOR_ELT(xSD, j))[0].r = NA_REAL;
+          COMPLEX(VECTOR_ELT(xSD, j))[0].i = NA_REAL;
+          break;
```

Comment:
IIUC this is coming from a join, so we won't be able to cover it yet.

Minimal example to trigger the `INTSXP` branch of this `switch` (adapted test 1283.3):

```
dt1 <- data.table(a=rep(1:2, each=2), c=c(7L, 2L, 4L, 8L))
dt2 <- data.table(b=rep(2:3), c=c(2L, 6L), d=c(20L, 8L))
dt1[dt2, on = c(a = 'b'), .(a=a), by=.EACHI]
```

I believe it's because `j` uses a column from the LHS of `on`.

But joins aren't available for complex yet...